docker/fetch_normalization_data.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python
+
+"""
+Fetches normalization data.
+
+Usage:
+
+    python docker/fetch_normalization_data.py
+
+"""
+
+import argparse
+import datetime
+import os
+import sys
+
+import psycopg2
+import requests
+
+
+# NOTE(willkg): These platforms come from the -prod data.
+PLATFORMS = ['Mac OS X', 'Linux', 'Windows', 'Unknown']
+
+
+def fetch_data_from_api(endpoint, product, version, platform, start_date, end_date):
+    """Fetches data from API"""
+    req = requests.get(endpoint, params={
+        'end_date': end_date,
+        'start_date': start_date,
+        'product': product,
+        'platforms': platform,
+        'versions': version,
+    })
+    if req.status_code != 200:
+        print('Error retrieving ADI: %s' % req.content)
+        return []
+
+    data = req.json()
+    return data['hits']
+
+
+def fetch_versions(conn, product_name):
+    """Fetch the version strings for a product from product_versions table"""
+
+    cursor = conn.cursor()
+    cursor.execute("""
+        SELECT version_string
+        FROM product_versions
+        WHERE
+            product_name=%s
+    """, (product_name,))
+    # NOTE(willkg): We don't want versions that end with "b" because they're weird.
+    versions = set([record[0] for record in cursor if not record[0].endswith('b')])
```

Comment:
This is curious. It seems like the API call returns all the data for versions with the specified prefix. So if we pass in something like 55.0b, then it returns data for 55.0b1, 55.0b2, ... and that gets messy.

https://crash-stats.mozilla.com/api/ADI/?end_date=2017-08-10&platforms=Mac%20OS%20X&product=Firefox&start_date=2017-08-03&versions=55.0b

So what I do here is skip that version because I don't know what else to do with it and the data from the API doesn't include it anyhow.

Is that the right thing to do? Is there a better way to deal with it?